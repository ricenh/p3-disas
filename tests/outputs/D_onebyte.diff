--- outputs/D_onebyte.txt	2023-11-02 16:25:02.064133059 -0400
+++ expected/D_onebyte.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,7 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of executable contents:
+  0x100:                               | .pos 0x100 code
+  0x100:                               | _start:
+  0x100: 00                            |   halt
+  0x101: 10                            |   nop
+  0x102: 90                            |   ret
+
