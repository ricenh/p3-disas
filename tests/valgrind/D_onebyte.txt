==1076159== Memcheck, a memory error detector
==1076159== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1076159== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1076159== Command: ../y86 -d inputs/onebyte.o
==1076159== 
Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 00                            |   halt
  0x101: 10                            |   nop
  0x102: 90                            |   ret

==1076159== 
==1076159== HEAP SUMMARY:
==1076159==     in use at exit: 0 bytes in 0 blocks
==1076159==   total heap usage: 4 allocs, 4 frees, 20,952 bytes allocated
==1076159== 
==1076159== All heap blocks were freed -- no leaks are possible
==1076159== 
==1076159== For lists of detected and suppressed errors, rerun with: -s
==1076159== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
