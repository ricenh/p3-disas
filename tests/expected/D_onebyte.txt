Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 00                            |   halt
  0x101: 10                            |   nop
  0x102: 90                            |   ret

